lib/credentials/dynamic.py

```diff
@@ -180,6 +181,7 @@ def generate(self, snapshot: Optional[str] = None, **kwargs):
                 security_class=self.security_class,
                 cred_type=self.cred_type,
             )
+            self.cred_type = self._generated_credential.cred_type
         else:
             raise CredentialError(f"Invalid generated value: {generated_value}. Expected a string or bytes.")
```

Comment:
Should the message be "Invalid generated value: {generated_value}. Expected a Credential, or a string or bytes."